app/clustering.py

```diff
@@ -0,0 +1,270 @@
+            Input data for clustering.
+            sample_weight (Optional[List[float]]):
+            Sample weights to influence clustering.
+
+        Returns:
+            ndarray: Cluster labels for each data point.
+
+        Example usage:
+            ```python
+            labels = dbscan.cluster(data)
+            print(labels)
+            ```
+        """
+        return self.model.fit_predict(data, sample_weight=sample_weight)
+
+
+class CAgglomerativeClustering:
+    """
+    Utility class for Agglomerative Clustering.
+
+    This class simplifies the process of using the Agglomerative Clustering
+    algorithm from scikit-learn, providing a wrapper around its implementation.
+    """
+
+    def __init__(
+        self,
+        n_clusters: int,
+        metric: str,
+        memory: Optional[str],
+        connectivity: Optional[list[Any]],
+        compute_full_tree: Union[Literal["auto"], bool],
+        linkage: Literal["ward", "complete", "average", "single"],
+        distance_threshold: Optional[float],
+        compute_distances: bool,
+    ) -> None:
+        """
+        Initialize the CAgglomerativeClustering class.
+
+        Args:
+            n_clusters (int): The number of clusters to find.
+            metric (str): The metric to use for computing distances.
+            memory (Optional[str]):
+            Cache to use for storing intermediate results.
+            connectivity (Optional[list[Any]]):
+            Connectivity matrix for samples.
+            compute_full_tree (Union[Literal["auto"], bool]):
+            Whether to compute the full tree.
+            linkage (Literal["ward", "complete", "average", "single"]):
+            Linkage criterion to use.
+            distance_threshold (Optional[float]):
+            Threshold for merging clusters.
+            compute_distances (bool):
+            Whether to compute distances between clusters.
+
+        Example usage:
+            ```python
+            from numpy import array
+            from your_module import CAgglomerativeClustering
+
+            data = array([[1, 2], [1, 4], [1, 0], [10, 2], [10, 4], [10, 0]])
+            agglomerative = CAgglomerativeClustering(
+                n_clusters=2, metric="euclidean", memory=None,
+                connectivity=None, compute_full_tree="auto",
+                linkage="ward", distance_threshold=None,
+                compute_distances=False
+            )
+            labels = agglomerative.cluster(data)
+            print(labels)
+            ```
+        """
+        self.model = AgglomerativeClustering(
+            n_clusters=n_clusters,
+            metric=metric,
+            memory=memory,
+            connectivity=connectivity,
+            compute_full_tree=compute_full_tree,
+            linkage=linkage,
+            distance_threshold=distance_threshold,
+            compute_distances=compute_distances,
+        )
+
+    def cluster(self, data: ndarray) -> ndarray:
```

Comment:
this method does not take sample_weights as other cluster methods, there more ways how to solve this:
1. all methods would take the argument and it can be None, when not None in this method, you can raise NotImplementedError, because this algorithm does not support it
2. do not use sample_weights in no custom implementation of clustering